src/run.py

```diff
@@ -0,0 +1,170 @@
+    targz_filename = Path("tmp")
+    with open(targz_filename, "wb") as file:
+        file.write(response.content)
+
+    # Open as a tarfile
+    tar = tarfile.open(targz_filename, "r:gz")
+    filenames = tar.getnames()
+    tar.extractall(path=Path("../data/"))
+    tar.close()
+    os.remove(targz_filename)
+    directory_name = Path("../data/" + filenames[0])
+    return directory_name
+
+
+def open_local_file(file_path):
+    """Open local source files"""
+    directory_name = Path("../data/" + file_path)
+    return directory_name
+
+
+def get_bkg_and_signal(directory_name, model_point):
+    """Load background and signal"""
+
+    bkgonly_path = directory_name / Path("BkgOnly.json")
+    signal_path = directory_name / Path("patchset.json")
+
+    if bkgonly_path.exists() and signal_path.exists():
+        background_only = json.load(open(bkgonly_path))
+        patchset = pyhf.PatchSet(json.load(open(signal_path)))
+        signal_patch = patchset[model_point]
+    elif bkgonly_path.exists():
+        background_only = json.load(open(bkgonly_path))
+        signal_patch = None
+    else:
+        json_file = open(directory_name.glob('*.json')[0])
+        background_only = json.load(json_file)
+        signal_patch = None
+
+    return background_only, signal_patch
+
+
+def calculate_CLs(bkgonly_json, signal_patch_json):
+    """
+    Calculate the observed CLs and the expected CLs band from a background only
+    and signal patch.
+
+    Args:
+        bkgonly_json: The JSON for the background only model
+        signal_patch_json: The JSON Patch for the signal model
+
+    Returns:
+        CLs_obs: The observed CLs value
+        CLs_exp: List of the expected CLs value band
+    """
+    workspace = pyhf.workspace.Workspace(bkgonly_json)
+    if signal_patch_json:
+        model = workspace.model(
+            measurement_name=None,
+            patches=[signal_patch_json],
+            modifier_settings={
+                "normsys": {"interpcode": "code4"},
+                "histosys": {"interpcode": "code4p"},
+            },
+        )
+    else:
+        model = workspace.model()
+
+    result = pyhf.infer.hypotest(
+        1.0, workspace.data(model), model, qtilde=True, return_expected_set=True
+    )
+    if isinstance(pyhf.tensorlib, pyhf.tensor.pytorch_backend):
+        return result[0].tolist()[0], result[-1].tolist()
+    else:
+        return result[0].tolist()[0], result[-1].ravel().tolist()
+
+def end(directory_name):
+    shutil.rmtree(directory_name)
+
+
+@click.command()
+@click.option(
+    "-b",
+    "--backend",
+    "backend",
+    help="Name of the pyhf backend to run with.",
+    default="numpy",
+    required=False,
+)
+@click.option(
+    "-p",
+    "--path",
+    "path",
+    help="Local path of workspace.",
+    default=None,
+    required=False,
+)
+@click.option(
+    "-u",
+    "--url",
+    "url",
+    help="Online data link.",
+    default="https://www.hepdata.net/record/resource/1267798?view=true",
```

Comment:
```suggestion
```
Probably don't want a default here.

Reply:
So, we don't want to use a default value for URL or path? The reason why I add default values here is that I just hope that I can run `run.py` in Pycharm without using a command line.

Reply:
Default values are added if they make sense for the end user, which it doesn't as we want to be able to run on an arbitrary file. Can't you just run using the CLI in PyCharm's terminal emulator?